Helpers.gs

```diff
@@ -502,43 +527,41 @@ function checkSkipEvent(event, icalEvent){
  * @param {Calendar.Event} recEvent - The event instance to process
  */
 function processEventInstance(recEvent){
-  Logger.log("ID: " + recEvent.extendedProperties.private["id"] + " | Date: "+ recEvent.recurringEventId.substring(0,10));
-  var recIDStart = new Date(recEvent.recurringEventId);
-  recIDStart = new ICAL.Time.fromJSDate(recIDStart, true);
-
-  var calendarEvents = Calendar.Events.list(targetCalendarId, 
-    { timeZone : "etc/GMT", 
-      singleEvents : true, 
+  Logger.log("ID: " + recEvent.extendedProperties.private["id"] + " | Date: "+ recEvent.recurringEventId);
+  
+  var eventInstanceToPatch = Calendar.Events.list(targetCalendarId, 
+    { singleEvents : true,
       privateExtendedProperty : "fromGAS=true", 
-      privateExtendedProperty : "id=" + recEvent.extendedProperties.private['id']
+      privateExtendedProperty : "rec-id=" + recEvent.extendedProperties.private["id"] + "_" + recEvent.recurringEventId
     }).items;
 
-  Logger.log("Found " + calendarEvents.length + " possible instances");
-  var eventInstanceToPatch = calendarEvents.filter(function(item){
-    try{
-      var origStart = item.originalStartTime.dateTime || item.originalStartTime.date;
-      var instanceStart = new ICAL.Time.fromString(origStart);
-
-      return (instanceStart.compare(recIDStart) == 0);
-    }catch(e){
-      Logger.log("Error: " + e);
-      return 0; 
+  if (eventInstanceToPatch.length == 0){
+    if (recEvent.recurringEventId.length == 10){
+      recEvent.recurringEventId += "T00:00:00Z";
     }
-  });
+    eventInstanceToPatch = Calendar.Events.list(targetCalendarId, 
+      { singleEvents : true,
+        orderBy : "startTime",
+        maxResults: 1,
+        timeMin : recEvent.recurringEventId,
+        privateExtendedProperty : "fromGAS=true", 
+        privateExtendedProperty : "id=" + recEvent.extendedProperties.private["id"]
+      }).items;
+  }
 
-  if (eventInstanceToPatch.length == 0){
-    Logger.log("No Instance matched, adding as new event!");
+  if (eventInstanceToPatch.length == 1){
     try{
-      Calendar.Events.insert(recEvent, targetCalendarId);
+      Logger.log("Updating existing event instance");
+      Calendar.Events.update(recEvent, targetCalendarId, eventInstanceToPatch[0].id);
     }
     catch(error){
       Logger.log(error); 
     }
   }
   else{
+    Logger.log("No Instance matched, adding as new event!");
     try{
```

Comment:
Don't know if we also want to replace this and the `try` above (for `Events.update`) (and maybe more) with `callWithBackoff` like you did below

Reply:
I guess we can add it to all api calls.